src/plugins/intel_cpu/src/nodes/random_uniform.cpp

```diff
@@ -399,135 +612,284 @@ std::pair<uint64_t, uint64_t> RandomUniform::computePhilox(void* out, size_t out
 
-    if (selectedPrimitiveDesc) {
-        if (selectedPrimitiveDesc->getConfig().outConfs[0].getMemDesc()->getPrecision() != ov::element::u8) {
-            str_type += "_" + std::string(selectedPrimitiveDesc->getConfig().outConfs[0].getMemDesc()->getPrecision().get_type_name());
-        } else {
-            str_type += "_I8";
+    uint32_t mersenne_state_ptr[MERSENNE_STATE_N];
+    auto output_byte_ptr = reinterpret_cast<uint8_t*>(out);
+    initial_mersenne_state(mersenne_state_ptr, m_global_seed);
+
+    if (m_jit_kernel) {
+#if defined(OPENVINO_ARCH_X86_64)
+        for (uint64_t i = 0; i < state_regenerations_required; ++i) {
+            next_mersenne_state(mersenne_state_ptr);
+            parallel_nt(m_threads_num, [&](const int ithr, const int nthr) {
+                kernel::random_uniform::MersenneTwisterGeneratorCallArgs args;
+                auto& params = m_mersenne_twister_thread_params[ithr];
+                args.min_ptr                = &m_min_val;
+                args.range_ptr              = &m_range_val;
+                args.max_output_idx         = output_elements_count;
+                args.state_accesses_count   = params.state_accesses_count;
+                args.state_ptr              = mersenne_state_ptr + params.src_start_idx;
+                args.dst_ptr                = output_byte_ptr + params.dst_start_idx + i * byte_offset;
+                args.output_idx             = params.dst_start_idx + i * MERSENNE_STATE_N;
+                args.elements_to_generate   = static_cast<int64_t>(
+                                                std::min(
+                                                    static_cast<uint64_t>(m_uint_storage_capacity_per_thread) * args.state_accesses_count,
+                                                    args.max_output_idx - args.output_idx));
+
+                if (args.output_idx >= args.max_output_idx) {
+                    return;
+                }
+
+                // For loop could not be inside the kernel as I ran out of Reg64s available
+                for (uint64_t j = 0; j < args.state_accesses_count; ++j) {
+                    (*m_jit_kernel)(&args);
+
+                    args.elements_to_generate =  std::max(
+                                                    args.elements_to_generate - static_cast<int64_t>(m_uint_storage_capacity_per_thread),
+                                                    static_cast<int64_t>(0));
+                    args.state_ptr = reinterpret_cast<uint8_t*>(args.state_ptr) + m_uint_storage_capacity_per_thread *  m_output_prc.size();
+                    args.dst_ptr = reinterpret_cast<uint8_t*>(args.dst_ptr) + m_uint_storage_capacity_per_thread * m_output_prc.size();
+                    args.output_idx += m_uint_storage_capacity_per_thread;
+                }
+            });
         }
-    }
+#endif // OPENVINO_ARCH_X86_64
+    } else {
+        const auto elements_generated_per_access = m_mersenne_twister_optimization_enabled ?
```

Comment:
Could you please compare performance of this approach versus kernel implementation?